KnownMimeTypes.tt

```diff
@@ -34,46 +37,57 @@ namespace MimeMapping
 		}
 
-        // entries where item 0 is the mime string, and the rest are file extensions
-        List<string[]> entries = new List<string[]>();
-
-        // Download and parse entries from the Apache file
-        using (var responseStream = WebRequest.CreateHttp(APACHE_URL).GetResponse().GetResponseStream())
-        using (var streamReader = new StreamReader(responseStream, Encoding.UTF8))
-        {
-            var response = streamReader.ReadToEnd();
-            var lines = response.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
-            foreach(var line in lines)
+		string GetPageContent(string url) 
+		{
+			using (var responseStream = WebRequest.CreateHttp(url).GetResponse().GetResponseStream())
+			using (var streamReader = new StreamReader(responseStream, Encoding.UTF8))
 			{
-				if (line[0] == '#')
-					continue;
-				var parts = line.Split(new char[0], StringSplitOptions.RemoveEmptyEntries);
-				if (parts.Length < 2)
-					continue;
-				entries.Add(parts);
+				return streamReader.ReadToEnd();				
 			}
 		}
 
-		// Download and parse entries from the H5BP nginx file
-		using (var responseStream = WebRequest.CreateHttp(NGINX_URL).GetResponse().GetResponseStream())
-		using (var streamReader = new StreamReader(responseStream, Encoding.UTF8))
+		IEnumerable<string[]> GetMimeTypesFromText(string url, Func<string, string[]> processLine)
+        {
+            var content = GetPageContent(url);
+			if (string.IsNullOrEmpty(content)) return Enumerable.Empty<string[]>();
+            var lines = content.Split(new[] {'\r', '\n'}, StringSplitOptions.RemoveEmptyEntries);
+            return lines.Select(processLine).Where(newKeyPairs => newKeyPairs != null).ToList();
+        }
+
+		IEnumerable<string[]> GetMimeDbCustomMimeTypeJson()
 		{
-			var response = streamReader.ReadToEnd();
-			var lines = response.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
-			foreach(var l in lines)
-			{
-				var line = l.Trim().TrimEnd(';');
-				if (line[0] == '#' || line[0] == '}')
-					continue;
-				if (line.StartsWith("types {"))
-					continue;
-				var parts = line.Split(new char[0], StringSplitOptions.RemoveEmptyEntries);
-				if (parts.Length < 2)
-					continue;
-				entries.Add(parts);
-			}
+			var content = GetPageContent(MIMEDB_URL);
+			if (string.IsNullOrEmpty(content)) return Enumerable.Empty<string[]>();
+			var resource = JObject.Parse(content);
+            return from mimeTypes in resource.Children() 
+                let mimeType = ((JProperty) mimeTypes).Name 
+                from mimeTypeProperties in mimeTypes.Children() 
+                from mimeTypeProperty in mimeTypeProperties.Children() 
+                where (mimeTypeProperty as JProperty)?.Name == "extensions" 
+                select new[]
+                {
+                    mimeType, 
+                    mimeTypeProperty.Values().First().Value<string>()
```

Comment:
Is it safe to only use the first entry here? I glanced over the json and the only notable missing extension I saw was in this entry:
```
  "text/yaml": {
    "extensions": ["yaml","yml"]
  },
```

Reply:
Good catch! Will address.